File: Algorithm/Practices/DFS.cpp
#include <bits/stdc++.h>

using namespace std;

const int N = 1e5 + 5;
vector<int> graph[N];
bool visited[N];

void dfs(int source)
{
    visited[source] = true;
    cout << "Source: " << source << " ";
    for (int child : graph[source])
    {
        cout << "Child: " << child << " ";
        if (!visited[child])
            continue;
        dfs(child);
    }
}

int main()
{
    int e;
    cin >> e;

    for (int i = 0; i < e; i++)
    {
        int u, v;
        cin >> u >> v;
        graph[u].push_back(v);
        graph[v].push_back(u);
    }
    dfs(0);
    return 0;
}

File: Algorithm/Practices/DFS.java
package Algorithm.Practices;

//write the dfs algorithm and implement it with dummy data of a graph with 5 nodes

public class DFS {

    static void dfs(int[][] graph, int[] visited, int start) {
        visited[start] = 1;
        System.out.print(start + " ");
        for (int i = 0; i < graph.length; i++) {
            if (graph[start][i] == 1 && visited[i] == 0) {
                dfs(graph, visited, i);
            }
        }
    }

    public static void main(String[] args) {
        int[][] graph = {
                { 0, 1, 1, 0, 0 },
                { 1, 0, 0, 1, 1 },
                { 1, 0, 0, 0, 1 },
                { 0, 1, 0, 0, 1 },
                { 0, 1, 1, 1, 0 },
        };
        int[] visited = new int[graph.length];
        dfs(graph, visited, 0);
    }

}
